### PR TITLE
Group by should require an aggregate

### LINKED ISSUE #2017: Parser should return error if no aggregate with group by time
A query like this should return an error from the parser:

```sql
select foo from bar group by time(300s)
```

It requires an aggregate function to be present in the select clause:

```sql
select mean(foo) from bar group by time(300s)
```

### PR DESCRIPTION
**Pull request by influxdb-denver-pair**
fixes #2017

### CODE REVIEW

File: httpd/handler.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ type Batch struct {
 // Return all the measurements from the given DB
 func (h *Handler) showMeasurements(db string, user *influxdb.User) ([]string, error) {
 	var measurements []string
-	results := h.server.ExecuteQuery(&influxql.Query{[]influxql.Statement{&influxql.ShowMeasurementsStatement{}}}, db, user)
+	results := h.server.ExecuteQuery(&influxql.Query{Statements: []influxql.Statement{&influxql.ShowMeasurementsStatement{}}}, db, user)
```
Comment (influxdb-denver-pair) [inline httpd/handler.go]: this was failing a vet check from a previous commit.

**Comment by influxdb-denver-pair** [comment]
Several tests were not valid due to not having an aggregate specified.  They have been updated as part of this PR.

File: influxql/parser.go [hunk 2]
```diff
@@ -585,6 +585,10 @@ func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, e
 		}
 	})
 
+	if d, _ := stmt.GroupByInterval(); stmt.IsRawQuery && d > 0 {
+		return nil, fmt.Errorf("group by needs at least one aggregate function")
```
Comment (influxdb-denver-pair) [inline influxql/parser.go]: I'm open to a better error message if someone has ideas.

Comment (otoolep) [inline influxql/parser.go]: Perhaps capitalize it, to make it clear it's influxql, and I think "requires" reads better. I.e. "GROUP BY requires at least one aggregate function"

**Comment by otoolep** [comment]
I have some minor feedback, which I think would improve the code. +1 on a green build.

**Pull request merged**